File: app/src/main/java/com/theandroiddev/mywins/data/models/SuccessDao.java
package com.theandroiddev.mywins.data.models;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by grazyna on 2017-11-18.
 */
@Dao
public interface SuccessDao {

    //TODO replace ASC DESC
    @Query("SELECT * FROM success WHERE title LIKE :searchTerm ORDER BY :sortType ASC")
    List<Success> getAllASC(String searchTerm, String sortType);

    @Query("SELECT * FROM success WHERE title LIKE :searchTerm ORDER BY :sortType DESC")
    List<Success> getAllDESC(String searchTerm, String sortType);

    @Query("SELECT * FROM success WHERE title LIKE :title LIMIT 1")
    Success findByName(String title);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(ArrayList<Success> successList);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void update(Success success);

    @Delete
    void delete(Success success);

    @Query("delete from success")
    void removeAll();

}

File: app/src/main/java/com/theandroiddev/mywins/data/models/SuccessDao.kt
package com.theandroiddev.mywins.data.models

import android.arch.persistence.room.*
import java.util.*

/**
 * Created by grazyna on 2017-11-18.
 */
@Dao
interface SuccessDao {

    //TODO replace ASC DESC
    @Query("SELECT * FROM success WHERE title LIKE :searchTerm ORDER BY :sortType ASC")
    fun getAllASC(searchTerm: String, sortType: String): List<Success>

    @Query("SELECT * FROM success WHERE title LIKE :searchTerm ORDER BY :sortType DESC")
    fun getAllDESC(searchTerm: String, sortType: String): List<Success>

    @Query("SELECT * FROM success WHERE title LIKE :title LIMIT 1")
    fun findByName(title: String): Success

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(successList: ArrayList<Success>)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun update(success: Success)

    @Delete
    fun delete(success: Success)

    @Query("delete from success")
    fun removeAll()

}
